### PR TITLE
Switched ohlc and candlestick charts to canvas

### PR DESCRIPTION
**Pull request by Ro4052**

### CODE REVIEW

**Comment by Ro4052** [comment]
I probably shouldn't have done this in one commit. There is also a fix for my d3fc canvas gridline component in here. It wasn't one that I'd noticed since we only use the full grid in canvas charts.

File: packages/perspective-viewer-d3fc/src/js/charts/ohlcCandle.js [hunk 1]
```diff
@@ -37,19 +37,20 @@ function ohlcCandle(seriesSvg) {
...
             .crossValue(d => d.crossValue)
             .openValue(d => d.openValue)
             .highValue(d => d.highValue)
             .lowValue(d => d.lowValue)
             .closeValue(d => d.closeValue)
-            .decorate(selection => {
-                selection.style("fill", d => (isUp(d) ? upColor(d.key) : downColor(d.key)));
-                selection.style("stroke", d => (isUp(d) ? upColor(d.key) : downColor(d.key)));
+            .decorate((context, d) => {
+                const color = isUp(d) ? upColor(d.key) : downColor(d.key);
```
Comment (mhooper-scottlogic) [inline packages/perspective-viewer-d3fc/src/js/charts/ohlcCandle.js:47]: Nice

File: packages/perspective-viewer-d3fc/src/js/charts/ohlcCandle.js [hunk 2]
```diff
@@ -90,7 +95,8 @@ function ohlcCandle(seriesSvg) {
             .chart(chart)
             .settings(settings)
             .xScale(xScale)
-            .yScale(yScale);
+            .yScale(yScale)
+            .canvas(true);
 
         const toolTip = nearbyTip()
             .settings(settings)
```
Comment (mhooper-scottlogic) [inline packages/perspective-viewer-d3fc/src/js/charts/ohlcCandle.js:102]:
```suggestion
            .settings(settings)
            .canvas(true)
```

Comment (mhooper-scottlogic) [reply packages/perspective-viewer-d3fc/src/js/charts/ohlcCandle.js:102]: I think this is needed to make tooltips work

**Review by mhooper-scottlogic — CHANGES_REQUESTED** [review]

File: packages/perspective-viewer-d3fc/src/js/charts/ohlcCandle.js (hunk 2 above)
Comment (Ro4052) [reply packages/perspective-viewer-d3fc/src/js/charts/ohlcCandle.js:102]: I don't know why I didn't check tooltips but I've confused by this suggestion.

Comment (Ro4052) [reply packages/perspective-viewer-d3fc/src/js/charts/ohlcCandle.js:102]: I'm not sure why it isn't working

Comment (Ro4052) [reply packages/perspective-viewer-d3fc/src/js/charts/ohlcCandle.js:102]: I see, it's the `canvas` property that needs to be set.

Comment (mhooper-scottlogic) [reply packages/perspective-viewer-d3fc/src/js/charts/ohlcCandle.js:102]: I think you just need to add .canvas(true). It worked for me:

const toolTip = nearbyTip()
            .settings(settings)
            .xScale(xScale)
            .yScale(yScale)
            .yValueName("closeValue")
            .canvas(true)
            .data(data);

File: packages/perspective-viewer-d3fc/src/js/charts/ohlcCandle.js (hunk 1 above)
Comment (Ro4052) [reply packages/perspective-viewer-d3fc/src/js/charts/ohlcCandle.js:47]: If we don't switch to canvas, maybe we should make this refactor anyway?

**Review by mhooper-scottlogic — APPROVED** [review]

File: packages/perspective-viewer-d3fc/src/js/charts/ohlcCandle.js (hunk 1 above)
Comment (mhooper-scottlogic) [reply packages/perspective-viewer-d3fc/src/js/charts/ohlcCandle.js:47]: Definitely

**Pull request merged**